src/marshaller/HipieDDL.js

```diff
@@ -917,6 +919,20 @@
         this._proxyMappings = {};
     }
 
+    Marshaller.prototype.commsDataLoaded = function () {
+        var dashboardLoadedArr = [];
+        var i = 0;
+        this.dashboardArray.forEach(function(dashboard) {
+            for (var ds in dashboard.datasources) {
+                dashboardLoadedArr[i] = dashboard.datasources[ds]._loadedCount;
+                i++;
+            }
+        });
+        return dashboardLoadedArr.every(function(element) {
+            return element > 0;
+        });
+    };
```

Comment:
rewrite this function to avoid double loop.  IOW the fist time you see a _loadedCount === 0 break and return false.